pouchDB/pouch.d.ts

```diff
@@ -1,230 +1,140 @@
+		query(fun: string, options?: Object, callback?: (error: Object, result: Object) => void): PouchDBPromise
+		query(fun: (doc: PouchDBDoc) => PouchDBDoc, options?: Object, callback?: (error: Object, result: Object) => void): PouchDBPromise
+		query(fun: (doc: Object) => Object, options?: Object, callback?: (error: Object, result: Object) => void): PouchDBPromise
+		viewCleanup(callback?: (error: Object, result: Object) => void): PouchDBPromise
+		info(callback?: (error: Object, result: Object) => void): PouchDBPromise
+		compact(options?: Object): PouchDBPromise
+		sync(remoteDB: string, options?: Object): PouchDBSync
+		revsDiff(diff: Object, callback?: (error: Object, result: Object) => void): PouchDBPromise
+	}
+	interface PouchDBDoc {
+		_id: string
+		_rev?: string
+	}
+	interface PouchDBAllDocsOpts {
+		include_docs?: boolean
+		conflicts?: boolean
+		attachments?: boolean
+		binary?: boolean
+		limit?: number
+		descending?: boolean
+		startkey?: string
+		endkey?: string
+		inclusive_end?: boolean
+		key?: string
+		keys?: Array<string>
+	}
+	interface PouchDBChangesOpts {
+		include_docs?: boolean
+		conflicts?: boolean
+		attachments?: boolean
+		binary?: boolean
+		limit?: number
+		descending?: boolean
+		live?: boolean
+		since?: any
+		timeout?: number
+		heartbeat?: number
+	}
+	interface PouchDBChangesOptsA extends PouchDBChangesOpts, PouchDBFilteringOptionsA { }
+	interface PouchDBChangesOptsB extends PouchDBChangesOpts, PouchDBFilteringOptionsB { }
+	interface PouchDBChanges extends PouchDBPromise {
+		on(handle: string, callback: (infoOrErr?: Object) => void): PouchDBChanges
+		cancel(): void
+	}
+	interface PouchDBReplicateOptsA extends PouchDBFilteringOptionsA {}
+	interface PouchDBReplicateOptsB extends PouchDBFilteringOptionsB {}
+	interface PouchDBFilteringOptions {
+		doc_ids?: string[]
+		query_params?: Object
+		view?: string
+		live?: boolean
+		retry?: boolean
+		back_off_function?: (delay: number) => number
+	}
+	interface PouchDBFilteringOptionsA extends PouchDBFilteringOptions {
+		filter?: string
+	}
+	interface PouchDBFilteringOptionsB extends PouchDBFilteringOptions {
+		filter?: (doc: Object) => boolean
+	}
+	interface PouchDBReplicateFromTo {
+		to(remoteDB: string, options?: PouchDBReplicateOptsA): PouchDBPromise
+		from(remoteDB: string, options?: PouchDBReplicateOptsA): PouchDBPromise
+		to(remoteDB: string, options?: PouchDBReplicateOptsB): PouchDBPromise
+		from(remoteDB: string, options?: PouchDBReplicateOptsB): PouchDBPromise
+	}
+	interface PouchDBSync extends PouchDBPromise {
+		on(handle: string, callback: (infoOrErr?: Object) => void): PouchDBSync
+		cancel(): void
+	}
+	interface PouchDBDebug {
+		enable(module: string): void
+		disable(): void
+	}
+	interface PouchDBPromise {
+		then(callback: (result: Object) => void): PouchDBPromise
+		catch(callback: (err: Object) => void): PouchDBPromise
+	}
+}
+
+declare var PouchDB: PouchDB.PouchDBFactory
+
+declare module "PouchDB" {
```

Comment:
module names are usually in lowercase. and I had to replace `export = PouchDB` by `export default PouchDB` to allow a default import ```import PouchDB from 'pouchdb';```